tests/unit/app/endpoints/test_ols.py

```diff
@@ -664,38 +664,6 @@ def test_conversation_request(
         assert len(response.conversation_id) == 0
 
 
-@pytest.mark.usefixtures("_load_config")
-@patch("ols.src.query_helpers.question_validator.QuestionValidator.validate_question")
-@patch("ols.src.query_helpers.docs_summarizer.DocsSummarizer.summarize")
-@patch("ols.config.conversation_cache.get")
-def test_conversation_request_dedup_ref_docs(
-    mock_conversation_cache_get,
-    mock_summarize,
-    mock_validate_question,
-    auth,
-):
-    """Test deduplication of referenced docs."""
-    mock_rag_chunk = [
-        RagChunk("text1", "url-b", "title-b"),
-        RagChunk("text2", "url-b", "title-b"),  # duplicate doc
-        RagChunk("text3", "url-a", "title-a"),
-    ]
-    mock_validate_question.return_value = True
-    mock_summarize.return_value = SummarizerResponse(
-        response="some response",
-        rag_chunks=mock_rag_chunk,
-        history_truncated=False,
-    )
-    llm_request = LLMRequest(query="some query")
-    response = ols.conversation_request(llm_request, auth)
-
-    assert len(response.referenced_documents) == 2
-    assert response.referenced_documents[0].docs_url == "url-b"
```

Comment:
It is tested in the new test.

Reply:
What exactly do you think is missing, because the current test tests the order as well.

Reply:
So you are afraid that it is passing because the input is already sorted?
This is not the case.

However if I add sorting into tested logic, the test will pass, but the behavior will be different. That is a problem with only two values/docs (the original test suffers from the same).

I've added a third to ensure there is no hidden sorting.